### PR TITLE
Expose the output_type option when streaming, and allow both stdout/stderr to be captured

### PR DESCRIPTION
**Pull request by tjgalvin**
This is an untested PR, but one I hope is fairly straightforward. I have a container whose program within it only outputs on the stderr. When streaming I am via the `Client.execute` method I am unable to capture these messages and log them. 

This PR aims to:
- expose the output_type option in execute (I have called it stream_type for clarity in the call into execute)
- ability to stream both the stdout and stderr. 
?
Would someone be able to give this a look over and test?

### CODE REVIEW

File: spython/utils/terminal.py [hunk 1]
```diff
@@ -122,12 +122,14 @@ def stream_command(
     sudo_options: string or list of strings that will be passed as options to sudo
 
     """
-    if output_type not in ["stdout", "stderr"]:
+    if output_type not in ["stdout", "stderr", "both"]:
         bot.exit("Invalid output type %s. Must be stderr or stdout." % output_type)
```
Comment (vsoch) [inline spython/utils/terminal.py:126]:
```suggestion
        bot.exit("Invalid output type %s. Must be stderr, stdout, or both" % output_type)
```

**Review by vsoch — COMMENTED** [review]
A small change is OK! Could you please ensure that:

- you've run black and linting
- bump the version in version.py
- add a note to the CHANGELOG with the corresponding version

Thank you!

**Comment by tjgalvin** [comment]
Alright, I believe I have run black on the two changes, bumped the version and added to the change log. 

Please do let me know if there is anything else that comes to mind. This was a very quick read for me, and I have not explicitly tested this new code path. It is small enough where I think it is OK.

**Comment by vsoch** [comment]
> I have not explicitly tested this new code path

if you are doing this for your own use and not adding a test here, I do need to ask you to at least test your local case (and show the result here).

For black, I recommend we pin to 23.3.0 (before the big change that produced the error and re-run).

**Comment by tjgalvin** [comment]
I had added a unit test to the test suite that attempts to verify this (and future proof things).

**Comment by tjgalvin** [comment]
And after patching my fork into my environment and testing it seems to be working as expected. I now have logs where there were no logs before.

**Pull request merged**